cmd/cosign/cli/attach/sig.go

```diff
@@ -106,12 +106,15 @@ func SignatureCmd(ctx context.Context, regOpts options.RegistryOptions, sigRef,
 		}
 
 		var localCosignPayload cosign.LocalSignedPayload
```

Comment:
This was intentional, to take in the output of `cosign sign --bundle path.txt ...`. The output is a "bundle", not a "rekor bundle".

Unfortunately we messed up with the naming, since we called this flag `rekor-response`. (For context, this is because we're trying to not use "bundle" anymore because we're working on support for a newer "bundle" format that will supersede the existing Cosign format).

Can we instead try to unmarshal this in two ways:
1. Leaving the current way, as a `cosign.LocalSignedPayload` and extracting `.Bundle`
2. To a [`bundle.RekorBundle`](https://github.com/sigstore/cosign/blob/e678426b3d524f15cab553e57eb6900f8280fb62/pkg/cosign/bundle/rekor.go#L23-L26)

And if both fail or either returns nil, then fail.

You'll also want to update the documentation of the flag - https://github.com/sigstore/cosign/blob/main/cmd/cosign/cli/options/attach.go#L61

Reply:
> This was intentional, to take in the output of cosign sign --bundle path.txt .... The output is a "bundle", not a "rekor bundle".

Basically, there is no such flag `bundle` in the `cosign sign`. Apart from that, what i am guessing is the intention behind adding this flag  was to get the output of the `bundle`, which contain:
- rekor bundle
- signature
- public key or certificate

At present we use the below command to externally attach signature in the image:
`cosign attach signature --payload payload.json --signature payload.json.base64.sig --rekor-response rekor_bundle.json $IMAGE `

So, it would be instead:
`cosign attach signature --payload payload.json --signature payload.json.base64.sig --rekor-bundle  rekor_bundle.json $IMAGE `

Reply:
Intentionally, we wanted to stop using "bundle" in flag names due to an upcoming change to add support for a newer bundle format. I would prefer to leave the flag name as-is and document that we accept either a bundle persisted from `cosign sign --bundle` or a rekor response formatted like that struct.